### PR TITLE
remove core/ from include path

### PR DESCRIPTION
**Pull request by karroffel**
core/ and core/math have been in the default include path, this can
cause trouble when a direction has files with the same basename as
files in core/ or core/math. (see #10071)

In order to fix such future problems all core/... includes need to
be explicit, preventing the compiler *not* including core files when
files with same basename are present.

### CODE REVIEW

File: core/array.cpp [hunk 1]
```diff
@@ -27,12 +27,12 @@
 /* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
-#include "array.h"
+#include "core/array.h"
```
Comment (akien-mga) [inline core/array.cpp]: As mentioned on IRC, in such case it should likely be kept as `array.h` to match the current .cpp.

**Comment by akien-mga** [comment]
I redid some of those changes manually in my recent cleanup, so... :p

**Comment by karroffel** [comment]
wheeey, will still be busy a while before coming back to this

**Comment by akien-mga** [comment]
Closing for now as it needs to be completely redone. I think it might be worth discussing applying the same changes for `editor` and `drivers`, to ensure we have consistent includes everywhere.

**Pull request closed without merging**